tap_github/client.py

```diff
@@ -77,11 +80,19 @@ def get_next_page_token(
         if not results:
             return None
 
-        if self.replication_key in ["starred_at", "created_at"]:
-            parsed_request_url = urlparse(response.request.url)
-            since = parse_qs(str(parsed_request_url.query))["since"][0]
-            if since and (parse(results[-1][self.replication_key]) < parse(since)):
-                return None
+        # Unfortunately endpoints such as /starred, /stargazers, /events and /pulls do not support
+        # the "since" parameter out of the box. So we use a workaround here to exit early.
```

Comment:
```suggestion
        # the "since" parameter out of the box. So we use a workaround here to exit early.
        # For such streams, we sort by descending dates (most recent first), and paginate
        # "back in time" until we reach records before our "since" parameter.
```